### DISCUSSION TITLE
How to create a new Mail provider?

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#14398] [Archive]
In the documentation:

> What to use to send emails. One of `sendmail`, `smtp`, `mailgun`, `ses`.

https://docs.directus.io/self-hosted/config-options/#email

But how we can create our own provider?
We want to use the [SendGrid](https://sendgrid.com) ([sendgrid/sendgrid-nodejs](https://github.com/sendgrid/sendgrid-nodejs)).

Thanks!

**Comment by abdonrd** [comment]
Oh, I think we can't do it with a PR:

https://github.com/directus/directus/blob/main/api/src/mailer.ts

**Comment by br41nslug** [comment]
This is indeed currently not possible to extend like extensions, however you can use the sendgrid service using the regular SMTP transport https://docs.sendgrid.com/for-developers/sending-email/getting-started-smtp

**Reply by abdonrd** [reply]
Thanks @br41nslug! We will try it!
